src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -70,6 +79,22 @@ public static bool IsXmsPageableOrArrayResponseOperation(Operation op, ServiceDe
             return false;
         }
 
+        // determine if the operation is xms pageable or returns an object of array type
+        public static bool IsXmsPageableOrArrayResponseOperation(Operation op, ServiceDefinition entity)
+        {
+            if (IsXmsPageableOperation(op))
```

Comment:
should we return (IsXmsPageableOperation(op) || IsArrayResponseOperation(op, entity)) ?